include/time_series/internal/base.hxx

```diff
@@ -213,3 +234,13 @@ size_t TimeSeriesBase<P, T>::max_length()
     read_indexes();
     return this->history_elements_ptr_->size();
 }
+
+template <typename P, typename T>
+void TimeSeriesBase<P, T>::monitor_signal()
+{
+    while (!signal_handler::SignalHandler::has_received_sigint())
+    {
+        real_time_tools::Timer::sleep_ms(10);
```

Comment:
maybe use a constant variable for 10

Reply:
Done. I also increased the duration to 100ms, I think that should be enough.